Remake static folder files to follow new theme

Hujim0 opened the pull request:
Modified form.html and result.html files to follow new theme inspired by [this web site](https://musicforprogramming.net/)

updated tailwind version to 4

!!!don't tested fully working app!!!
(i can't figure out how to launch a real world example)

Hujim0 commented:
![{816820B6-7C6E-46ED-A5F8-9B5F3F599FF7}](https://github.com/user-attachments/assets/85ae52e7-1bdd-42d0-adaa-73bc4668a446)
![{0BE0DD07-28A4-46DF-9F3D-00D90A0B2847}](https://github.com/user-attachments/assets/ee085881-10a1-4dd4-b1cb-382df9e77129)
![{F02FC1CE-F9C5-406B-A088-EA066DBF2BBB}](https://github.com/user-attachments/assets/2babea34-a241-42c5-9243-fe02d9463666)
![{7C716A91-4AFC-4413-876C-77893776FFDB}](https://github.com/user-attachments/assets/bbbcd50f-abbf-4469-8be9-82ab1f422e53)
![{7EFC362D-A50D-45C1-A15F-D9DFCF2E58D7}](https://github.com/user-attachments/assets/264d1af1-bbf4-4b67-8ed8-a891a8326d82)
![{01BA7B2B-E898-4B22-80F2-17AD43FFF8EF}](https://github.com/user-attachments/assets/84f3dc8b-f1f1-4204-91bb-47c295178f2f)

ryukzak reviewed (changes requested):
A few things:

1. Invert doesn't make sense if it resets on page reload. Save it in a local store and it will look nice.

2. Make the default font a little bit smaller. Around 70%.

3. Make the window width. Right now the report is unreadable. The columns should have the same widths.

4. Selection color.

Hujim0 commented:
Fixed everything except for the 
> 5. For some reason config don't upload correctly.

Don't know what can be an issue, sending the request is the same from the previous version of static htmls

maybe there can be some newline escaping issue when i switched off auto line breaks?

Hujim0 commented:
now report.html looks like this

![{195FE02F-7486-4136-9C7A-AA3BE1AE5DDA}](https://github.com/user-attachments/assets/b9eb42b5-424d-4512-9be5-15828fb8eae6)


and the selection color is just white
![{3C14EF31-6A5A-4AEF-8AE5-0B5878CD9090}](https://github.com/user-attachments/assets/cb5d3755-7ebf-43b3-b539-d86a35a52e81)

ryukzak commented:
In report still don't use whole windows width. Colors in headers should be consitence (on form -- only gray in `/* header */`, which I like more, on report -- many colors.

Hujim0 commented:
I heard that you said you want an grey headers, so i made them

I had an idea how to style author and variant strings, can you please choose what looks better in your opinion?

![{6AEDED4E-60E8-4D92-BB4E-C1EDCDC439ED}](https://github.com/user-attachments/assets/77d6e45f-1a18-4fbb-94eb-b198cd80fba3)
![{BF2D1D15-E63A-4480-90ED-00337063DD80}](https://github.com/user-attachments/assets/4d01f558-c746-4607-a6c1-ee95bc89d6ee)

ryukzak commented:
From the code perspective. Very long base64 string is not a good idea due to a lot of tooling issues. Use it from CDN, or we can just use anonymous pro everywhere.

Hujim0 commented:
> at about uploading issue and page

page width is now consistent

Hujim0 commented:
> 3\. On how it works -- who knows, who knows... But after I click back, I see only single line in text areas.

could you please show me some example data of something cause i really can't understand the issue 😭 

if anything, we could revert the change with white spaces to how it was before this whole restyling and call it a day

Hujim0 commented:
this is how it looks with empty data
![{671D6DF0-8C67-4A94-AA06-6F631D13802C}](https://github.com/user-attachments/assets/89aa5272-d091-4243-ad9f-437613725353)

everything except for the comment above is done i think

ryukzak commented:
1. It will be much simpler if you debug it by yourself. You can compare the master and your version's behaviour. It looks more straightforward, especially when you have full access to the source code. If it will raise problems, ask in the Wrench 3 topic -- someone has experience with this project.

2. On width. I really like the current (master) behaviour, where I have two columns on the full browser width.

Hujim0 commented:
okay, i tested with real data and here's what i got

![{CF343857-7706-4845-9FA4-496E3F8AB504}](https://github.com/user-attachments/assets/aef66820-db96-41ed-9368-4c6f085e2bad)


i think we should make the code and config sections as accordions too.

and empty comment case should be handled

Hujim0 commented:
Tho i saw no issues with config

![{F6C02E79-FACF-46E4-8902-3C67D8E160EF}](https://github.com/user-attachments/assets/0d106cda-4253-41b4-b62b-ab3e892e29f2)

Hujim0 commented:
found an issue with test_case_result

ryukzak commented:
1. Better and better. But width is not enough (Safari):

<img width="1657" alt="Screenshot 2025-03-06 at 00 42 12" src="https://github.com/user-attachments/assets/2c18c46f-1176-49ee-b452-861a233777c5" />

2. Is it possible to add `copy` button for asm/config text area? Looks like it is a common usecase.

3. Can you copy/paste link under the page title from form to report? Looks it nice have it their too, especially when we already took a place for invert button.

ryukzak commented:
And for the run simulation problem. Reproduce it only in Safari. For some reason, the indentation space changed. Here are the examples of the same texts, uploaded via Safari and via Firefox:


[Archive.zip](https://github.com/user-attachments/files/19099185/Archive.zip)

If I remove all classes for inputs, everything starts working.

Hujim0 commented:
> And for the run simulation problem. Reproduce it only in Safari. For some reason, the indentation space changed. Here are the examples of the same texts, uploaded via Safari and via Firefox:
> 
> [Archive.zip](https://github.com/user-attachments/files/19099185/Archive.zip)
> 
> If I remove all classes for inputs, everything starts working.

Thanks! Data helped a lot. I made sure that &nbsp (\u00A0) is replaced with just spaces. Hopefully it will resolve issue with safari. Its a pain to try run WebKit on windows or latest ubuntu

ryukzak commented on static/form.html:
```diff
@@ -1,187 +1,366 @@
+            <div class="mb-6">
+                <label for="asm" class="block grey">/* Assembler Code */</label>
+                <textarea id="asm" name="asm" required
+                    class="whitespace-nowrap overflow-x-auto subtle-hover bg-dark-gray mt-1 block w-full px-3 py-2 border border-zinc-400 rounded-md"></textarea>
```
Looks like it is a nbsp problem source. If I am mistaken, whitespace-pre will resolve the problem in a better way.

ryukzak commented on static/form.html:
```diff
@@ -1,187 +1,366 @@
+            <script>
+                function Submit() {
+                    let asmElement = document.getElementById("asm");
+                    asmElement.value = asmElement.value.replace(new RegExp("\u00A0", "g"), " ");
```
dirty hack.

ryukzak commented on static/result.html:
```diff
@@ -1,216 +1,527 @@
+            document.body.classList.add('inverted')
+        }
+
+        var copiedLinkCount = 0;
```
Let's simplify it. We don't need to count them.

ryukzak commented on static/result.html:
```diff
@@ -1,216 +1,527 @@
+        .vim-folded::after {
+            display: inline;
+            content: "\00A0··························································································································································";
+            white-space: nowrap;
```
Don't sure, but looks like you mix css and tailwind in case, when we can leave only tailwind.

ryukzak reviewed (commented):
Not a good idea to fix consequences instead of the problem (about nbsp). As usually, it hits in unexpected ways in other places, e.g., what you copy in the result page.

I will merge this PR right now but I hope you will fix the comments in it in another PR. Be aware, I add formatting check on the CI.

You can see what you do here: <http://wrench-edge.edu.swampbuds.me>
